pkg/reconciler/pipelinerun/pipelinespec/pipelinespec.go

```diff
@@ -57,10 +62,14 @@ func GetPipelineData(ctx context.Context, pipelineRun *v1beta1.PipelineRun, getP
 			pipelineMeta = pipeline.PipelineMetadata()
 			pipelineSpec = pipeline.PipelineSpec()
 		}
+		configSource = source
 	default:
 		return nil, nil, fmt.Errorf("pipelineRun %s not providing PipelineRef or PipelineSpec", pipelineRun.Name)
 	}
 
 	pipelineSpec.SetDefaults(ctx)
-	return &pipelineMeta, &pipelineSpec, nil
+	return &tresources.ResolvedObjectMeta{
```

Comment:
https://github.com/tektoncd/pipeline/pull/5397#discussion_r974607056 might be related. I think the main reason is that this helps us pass in all metadata about the resolved object in 1 struct rather than splitting into separate fields.

> Also, the taskRun reconciler package seems like an odd place for this struct to be defined in

I concur with that given the fact that it is used in both taskrun and pipelinerun reconciler, and this struct not only has *metav1.ObjectMeta but also has the source metadata. Do you have better suggestion on where we should define this struct if we go down that road? I am open to discussion here.

cc @wlynch

Reply:
Hmm, looks like there are many instances of the pipelinerun reconciler depending on taskrun reconciler's resources package :/

Maybe pkg/internal/resolved_meta.go?

Reply:
Thanks for the idea. Putting into internal SGTM. I put it under `pkg/internal/resolution/resolved_meta.go` file since the internal folder has subdirectories. PTAL. Let me know if this doesn't make sense or if you have other feedback. Many Thanks!